src/server/hotkey_collector.h

```diff
@@ -75,28 +81,48 @@ class hotkey_collector : public dsn::replication::replica_base
     void analyse_data();
     void handle_rpc(const dsn::replication::detect_hotkey_request &req,
                     /*out*/ dsn::replication::detect_hotkey_response &resp);
+    static int get_bucket_id(dsn::string_view data);
```

Comment:
you shouldn't make this function to be a member of class `hotkey_collector`

Reply:
it will be used in `hotkey_coarse_data_collector` and `hotkey_fine_data_collector`
I think it is a public method